test/e2e/support/world.js

```diff
@@ -1,25 +1,48 @@
 const fs = require('fs')
 const vm = require('vm')
 const path = require('path')
-const hasher = require('crypto').createHash
 const mkdirp = require('mkdirp')
-const _ = require('lodash')
+const rimraf = require('rimraf')
 const { setWorldConstructor } = require('cucumber')
 const Proxy = require('./proxy')
 
 class World {
   constructor () {
     this.proxy = new Proxy()
-    this.template = _.template(`process.env.CHROME_BIN = require('puppeteer').executablePath(); module.exports = function (config) {\n  config.set(\n    <%= content %>\n  );\n};`)
 
-    this.configFile = {
+    /**
+     * The current working directory path for all Karma commands.
+     * @type {string}
+     */
+    this.workDir = fs.realpathSync(__dirname)
+
+    /**
+     * The directory where all files generated during tests are stored.
+     * It is removed after each scenario.
+     * @type {string}
+     */
+    this.sandboxDir = path.join(this.workDir, 'sandbox')
+
+    /**
+     * Path to the final Karma config file.
+     * @type {string}
+     */
+    this.configFile = path.join(this.sandboxDir, 'karma.conf.js')
+
+    /**
+     * Absolute path to the Karma executable.
+     * @type {string}
+     */
+    this.karmaExecutable = fs.realpathSync(path.join(__dirname, '../../../bin/karma'))
+
+    this.config = {
       singleRun: true,
       reporters: ['dots'],
       frameworks: ['jasmine'],
-      basePath: __dirname,
+      basePath: this.workDir,
       colors: false,
-      __dirname: __dirname,
-      _resolve: (name) => path.resolve(__dirname, '..', 'support', name)
+      // TODO: Find a way to pass config overrides which does not require hacks
```

Comment:
Rather than TODO can you change this comment warn reader of the undesired state or possible outcome?  That works as both a heads up and as possible inspiration for improvement.

Reply:
@johnjbarton Done